### ISSUE TITLE
//bindings/python/pyiree/tf_interop:saved_model_test failing in OSS

### ISSUE THREAD

**Issue by GMNGeoffrey** [google/iree#131]
Bazel & Linux

```
Traceback (most recent call last):
  File "/usr/local/google/home/gcmn/.cache/bazel/_bazel_gcmn/c17d50ae90e3b490378de262a2ee1611/sandbox/linux-sandbox/1208/execroot/iree_core/bazel-out/k8-opt/bin/bindings/python/pyiree/tf_interop/saved_model_test.runfiles/iree_core/bindings/python/pyiree/tf_interop/saved_model_test.py", line 70, in testLoadSavedModelToXlaPipeline
    options = tf.saved_model.SaveOptions(save_debug_info=True)
AttributeError: module 'tensorflow._api.v1.compat.v2.saved_model' has no attribute 'SaveOptions'
```

This is passing internally. I already tried bumping the TF version, and that didn't fix it

**Comment by stellaraccident** [comment]
This is due to an incorrect version of TensorFlow (from July) being pip installed. I filed an internal bug about it and then closed it because it is WAI and following the install instructions exactly works.

The root cause is that somewhere around July 2019, the TF pip package started requiring a newer version of Pip than is installed on our Linux boxes (we had v18 and it requires 19.x). As a result, pip silently falls back to the last version that was compatible with what is installed on the system, which is quite old.

Here is the excerpt from the internal bug where we solved it:
---
Actually, I got myself out of the state. Here is what happened:

* I was originally using an out of date pip (v18).
* I installed via "pip3 install tf-nightly"
  * This installed 1.15 (I wasn't paying attention and I only noticed when some TF2 code had strange errors)
* I uninstalled "pip3 uninstall tf-nightly"
* Updated pip: "python3 -m pip install --upgrade pip"
* Re-installed tf-nightly, but I think it was using the cached version and installed 1.15 again.
* Finally passed an --upgrade command and it worked "pip3 install tf-nightly --upgrade"
---
TL;DR, run:
python3 -m pip install --upgrade pip  # Upgrades pip
pip3 install tf-nightly --upgrade

**Comment by GMNGeoffrey** [comment]
Thanks! I think this is also failing on the cbuild I'm iterating on. I guess there I should be pip installing tf-nightly? I'll see if this gets things working and add it to the linux docs

**Comment by GMNGeoffrey** [comment]
```
$ pip3 install tf-nightly --upgrade
Traceback (most recent call last):
  File "/usr/bin/pip3", line 11, in <module>
    sys.exit(main())
TypeError: 'module' object is not callable
```
:-/

I'll continue iterating

**Comment by GMNGeoffrey** [comment]
Hmm I've uninstalled and installed pip and tf-nightly a few times

`python3 -m pip install tf-nightly --upgrade --user` ended up working

and now I get different errors:

AttributeError: module 'pyiree.compiler' has no attribute 'tf_load_saved_model'

It looks like this has to do with TF being turned off

I tried:

`bazel test --copt="-D=iree_tensorflow true" //bindings/python/pyiree/tf_interop:saved_model_test`

which still doesn't work. Maybe I got the syntax wrong

**Comment by GMNGeoffrey** [comment]
I finally got this to work after running into issue #174 as well and having to downgrade to Bazel 1.1.0

For future people who may encounter this, you've got to make sure you have pip 19.x to get tf 2.x

I had to always use `python3 -m pip` rather than pip3 directly (which just fails for some reason).

First make sure you have pip version 19 `python3 -m pip --version`. I ran into some issues where it varied by user (different with sudo), so I'd recommend checking with the version with sudo as well (really I should probably have environments and such).

Then try `[sudo] python3 -m pip install tf-nightly --upgrade [--user]` (sudo or --user again depending on the user to install it for). If that command refuses to upgrade with something like "Requirement already up-to-date", make sure it says you already have version 2.x. 

Then 
```
bazel test --define=iree_tensorflow=true --test_output=errors //bindings/python/pyiree/tf_interop:saved_model_test
```
should work.
